hubspot/hubspot_test.go

```diff
@@ -0,0 +1,314 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hubspot
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"testing"
+)
+
+// setup sets up a test HTTP server along with a hubspot.Client that is
+// configured to talk to that test server.
+func setup() (client *Client, mux *http.ServeMux, teardown func()) {
+	mux = http.NewServeMux()
+	server := httptest.NewServer(mux)
+
+	client = NewClient("secret", server.Client())
+	url, _ := url.Parse(server.URL)
```

Comment:
Could you please tell me why do we need do it?

Reply:
The `server.URL` is pointed to a test server that we start and can use within the unit tests for emulating the HubSpot API. It allows us to redefine the `defaultBaseURL` which is equal to `https://api.hubapi.com`.